build.sh

```diff
@@ -97,65 +97,82 @@ cd EmulatorJS
-git pull
-cd "dist-scripts"
+compileStartPath="$PWD"
+for row in $(jq -r '.[] | @base64' ../cores.json); do
+    cd $compileStartPath
+
+    _jq() {
+        echo ${row} | base64 --decode | jq -r ${1}
+    }
+
+    name=`echo $(_jq '.') | jq -r '.name'`
+    repo=`echo $(_jq '.') | jq -r '.repo'`
+    license=`echo $(_jq '.') | jq -r '.license'`
+    buildpath=`echo $(_jq '.') | jq -r '.makeoptions.buildpath'`
+    makescript=`echo $(_jq '.') | jq -r '.makeoptions.makescript'`
+    arguments=`echo $(_jq '.') | jq -r '.makeoptions.arguments[] | @base64'`
+
+    argumentstring=""
+    for rowarg in $(echo "${arguments}"); do
+        argumentstring="$argumentstring `echo $rowarg | base64 --decode`"
+    done
+
+    echo "Starting compile of core $name"
+    echo "Working dir $PWD"
+
+    compileProject "$name" "$repo.git" "$buildpath" "$makescript" "$argumentstring"
+
+    # write JSON stanza for this core to disk
+    echo ${row} | base64 --decode > "./core.json"
+
+    if [ ! -z "$license" -a "$license" != " " ]; then
+        # license file is provided - copy it
+        echo "License file: $name/$license"
+        cp $name/$license "./license.txt"
+    fi
 
-mv core-temp/normal/*.bc ./
-emmake ./build-emulatorjs.sh emscripten clean no no
-rm -f *.bc
+    echo "Build wasm's for core $name"
+    cd "RetroArch"
+    git pull
+    cd "dist-scripts"
 
-mv core-temp/threads/*.bc ./
-emmake ./build-emulatorjs.sh emscripten clean yes no
-rm -f *.bc
+    mv core-temp/normal/*.bc ./
+    emmake ./build-emulatorjs.sh emscripten clean no no
+    rm -f *.bc
 
-mv core-temp/legacy/*.bc ./
-emmake ./build-emulatorjs.sh emscripten clean no yes
-rm -f *.bc
+    mv core-temp/threads/*.bc ./
+    emmake ./build-emulatorjs.sh emscripten clean yes no
+    rm -f *.bc
 
-mv core-temp/legacyThreads/*.bc ./
-emmake ./build-emulatorjs.sh emscripten clean yes yes
-rm -f *.bc
+    mv core-temp/legacy/*.bc ./
+    emmake ./build-emulatorjs.sh emscripten clean no yes
+    rm -f *.bc
+
+    mv core-temp/legacyThreads/*.bc ./
+    emmake ./build-emulatorjs.sh emscripten clean yes yes
+    rm -f *.bc
+
+    echo "Add support files to $name wasms"
+    cd $compileStartPath
+    if [ -f "EmulatorJS/data/cores/$name-wasm.data" ]; then
+        7z a -t7z EmulatorJS/data/cores/$name-wasm.data ./core.json ./license.txt ../build.json
+    fi
+
+    if [ -f "EmulatorJS/data/cores/$name-thread-wasm.data" ]; then
+        7z a -t7z EmulatorJS/data/cores/$name-thread-wasm.data ./core.json ./license.txt ../build.json
+    fi
+
+    if [ -f "EmulatorJS/data/cores/$name-legacy-wasm.data" ]; then
+        7z a -t7z EmulatorJS/data/cores/$name-legacy-wasm.data ./core.json ./license.txt ../build.json
+    fi
+
+    if [ -f "EmulatorJS/data/cores/$name-thread-legacy-wasm.data" ]; then
+        7z a -t7z EmulatorJS/data/cores/$name-thread-legacy-wasm.data ./core.json ./license.txt ../build.json
+    fi
+
+    # clean up
+    rm ./license.txt
+    rm core.json
+done
 
 cd "$initialPath"
```

Comment:
If no errors occurred, could we delete all the build files? Logs should always be kept.

Reply:
Cores now get copied to /output as they're created. Logs are stored in /output/logs. /compile is deleted at the end of the script.

Reply:
Sample state of the state of the repo after compilation completes. A single core was compiled for this example:
<img width="386" alt="Screenshot 2024-06-28 at 9 15 18 PM" src="https://github.com/EmulatorJS/build/assets/84688932/8d5d8c92-232e-4eb5-ae17-4ad8bf491438">